src/components/AuthContext.js

```diff
@@ -0,0 +1,15 @@
+import { createContext, useState } from "react";
+
+const AuthContext = createContext();
+
+export const AuthProvider = ({ children }) => {
+    const [token, setToken] = useState(localStorage.getItem('token') || null);
+
+    return (
+        <AuthContext.Provider value={{token, setToken}}>
+            {children}
+        </AuthContext.Provider>
+    );
+};
+
+export default AuthContext;
```

Comment:
Using createContext method to pass token data to all children components. Top level App component is wrapped within this AuthContext.Provider